docs/migration-to-NRI.md

```diff
@@ -0,0 +1,228 @@
+- For other policies a little bit more work is required and the policies need to be 'ported'. This can be done by just following the example of how balloons or topology-aware were converted.
+
+# Steps for an initial/basic migration test
+
+## Containerd
+
+Replace the containerd version in the system with 1.7 or newer version (NRI server not supported in older versions).
+
+Replace kubelet's --container-runtime-endpoint=/var/run/cri-resmgr/cri-resmgr.sock with --container-runtime-endpoint=/var/run/containerd/containerd.sock
+
+Replacing the runtime endpoint on a node that was setup using Kubeadm:
+```console 
+# Get the Kubelet args
+systemctl cat kubelet <- Look for: EnvironmentFile=/.../kubeadm-flags.env
+
+vim /.../kubeadm-flags.env
+  KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+
+vim /etc/sysconfig/kubelet 
+  KUBELET_EXTRA_ARGS= --container-runtime-endpoint=/var/run/containerd/containerd.sock <- Remember this aswell
+
+systemctl restart containerd
+systemctl restart kubelet
+```
+
+Edit the containerd config file and look for the section [plugins."io.containerd.nri.v1.nri"] and replace "disable = true" with "disable = false":
+```console
+vim /etc/containerd/config.toml
+```
+```toml
+[plugins."io.containerd.nri.v1.nri"]
+  disable = false
+  disable_connections = false
+  plugin_config_path = "/etc/nri/conf.d"
+  plugin_path = "/opt/nri/plugins"
+  plugin_registration_timeout = "5s"
+  plugin_request_timeout = "2s"
+  socket_path = "/var/run/nri/nri.sock"
+```
+
+## CRI-O
+
+Ensure that crio version 1.26.2 or newer is used.
+
+Replace kubelet's --container-runtime-endpoint=/var/run/cri-resmgr/cri-resmgr.sock with --container-runtime-endpoint=/var/run/crio/crio.sock
+
+Replacing the runtime endpoint on a node that was setup using Kubeadm:
+```console 
+# Get the Kubelet args
+systemctl cat kubelet <- Look for: EnvironmentFile=/.../kubeadm-flags.env
+
+vim /.../kubeadm-flags.env
+  KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/crio/crio.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+
+vim /etc/sysconfig/kubelet 
+  KUBELET_EXTRA_ARGS= --container-runtime-endpoint=/var/run/crio/crio.sock <- Remember this aswell
+
+systemctl restart crio
+systemctl restart kubelet
+```
+
+Edit the crio config and look for the "[crio.nri]" section, replace "enable_nri = false" with "enable_nri = true" and uncomment the NRI config lines:
+```console
+vim /etc/crio/crio.conf
+```
+
+```
+# CRI-O NRI configuration.
+[crio.nri]
+
+# Globally enable or disable NRI.
+enable_nri = true
+
+# NRI configuration file to use.
+nri_config_file = "/etc/nri/nri.conf"
+
+# NRI socket to listen on.
+nri_listen = "/var/run/nri/nri.sock"
+
+# NRI plugin directory to use.
+nri_plugin_dir = "/opt/nri/plugins"
+```
+
+## Build the NRI policies
+
+```console
+git clone https://github.com/jukkar/nri-resmgr.git -b preview
```

Comment:
The code was moved to
https://github.com/containers/nri-plugins
so this link can be changed to
```
git clone https://github.com/containers/nri-plugins
```